monai/handlers/dice_ce_metric.py

```diff
@@ -0,0 +1,63 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from monai.handlers.ignite_metric import IgniteMetric
+from monai.utils import MetricReduction
+from monai.losses import DiceCELoss
+from monai.metrics import LossMetric
+
+class DiceCEMetric(IgniteMetric):
+    """
+    Computes DiceCE score metric from full size Tensor and collects average over batch, class-channels, iterations.
+    """
+
+    def __init__(
+        self,
+        # include_background: bool = True,
+        reduction: MetricReduction | str = MetricReduction.MEAN,
+        # num_classes: int | None = None,
+        output_transform: Callable = lambda x: x,
+        save_details: bool = True,
+        *args,
+        **kwargs
+    ) -> None:
+        """
+
+        Args:
+            include_background: whether to include dice computation on the first channel of the predicted output.
+                Defaults to True.
+            reduction: define the mode to reduce metrics, will only execute reduction on `not-nan` values,
+                available reduction modes: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``,
+                ``"mean_channel"``, ``"sum_channel"``}, default to ``"mean"``. if "none", will not do reduction.
+            num_classes: number of input channels (always including the background). When this is None,
+                ``y_pred.shape[1]`` will be used. This option is useful when both ``y_pred`` and ``y`` are
+                single-channel class indices and the number of classes is not automatically inferred from data.
+            output_transform: callable to extract `y_pred` and `y` from `ignite.engine.state.output` then
+                construct `(y_pred, y)` pair, where `y_pred` and `y` can be `batch-first` Tensors or
+                lists of `channel-first` Tensors. the form of `(y_pred, y)` is required by the `update()`.
+                `engine.state` and `output_transform` inherit from the ignite concept:
+                https://pytorch.org/ignite/concepts.html#state, explanation and usage example are in the tutorial:
+                https://github.com/Project-MONAI/tutorials/blob/master/modules/batch_output_transform.ipynb.
+            save_details: whether to save metric computation details per image, for example: mean dice of every image.
+                default to True, will save to `engine.state.metric_details` dict with the metric name as key.
+            args: Arguments for the DiceCELoss
+
+        See also:
+            :py:meth:`monai.metrics.meandice.compute_dice`
+        """
+        # metric_fn = DiceMetric(include_background=include_background, reduction=reduction, num_classes=num_classes)
+        loss_function = DiceCELoss(*args, **kwargs)
+        metric_fn = LossMetric(loss_fn=loss_function, reduction=reduction, get_not_nans=False)
```

Comment:
thanks, do you think we can make a generic `LossMetricHandler` here, instead of hard-coding `DiceCELoss`, this class can take a user-initialized `loss_function` as input? 

currently it's achievable with the `IgniteMetric` https://github.com/Project-MONAI/model-zoo/blob/b245833e22544126f23925eac89da471562467a1/models/mednist_reg/configs/train.yaml#L180-L185 but I think `LossMetricHandler` would make the module logic easier to understand.

Reply:
@wyli Sure, I just added the code for that. The tests will be even more broken then. But in theory it works just fine, I am calling it with

```python
    all_val_metrics["val_mean_dice"] = IgniteLossMetric(
        loss_fn=loss_function_metric,
        output_transform=from_engine(["pred", "label"]),
    )
```

I called it IgniteLossMetric, but I can change it to LossMetricHandler if you prefer that.

Anyways I'm not even sure it we need this class, since it only has 3 actual lines of code. The same results can simply be achieved with the current code by doing it like this:

```python
    loss_function_metric = DiceCELoss(softmax=True, squared_pred=True)
    metric_fn = LossMetric(loss_fn=loss_function_metric, reduction="mean", get_not_nans=False)
    ignite_metric = IgniteMetric(metric_fn=metric_fn, output_transform=from_engine(["pred", "label"]), save_details=True)

    all_val_metrics = dict()
    all_val_metrics["val_mean_dice"] = ignite_metric

```
Only issue I see here is that this super non intuitive for a beginner wanting to create a new metric.. Maybe overloading the IgniteMetric to accept either metric_fn or loss_fn would be another idea?

Reply:
sure, IgniteLossMetric looks good to me.. the related metrics handlers are also simple https://github.com/Project-MONAI/MONAI/blob/dev/monai/handlers/mean_dice.py cc @Nic-Ma @ericspod

(please let me know if you need any help creating the unit tests.)

Reply:
Hi @matt3o. It's only 3 lines as you show to implement this but it would be good to reduce confusion. I like the idea of adding a `loss_fn` argument to `IgniteMetric` to wrap whatever is passed by it within a `LossMetric` object. 

One issue that would be worth fixing as well is that `IgniteMetric` is in fact a handler whereas `LossMetric` is an actual metric class. It's out of scope here but I'd suggest renaming `IgniteMetric` to `IgniteMetricHandler` or something that makes it clearer what the class is not a metric but a handler _for_ metrics.